src/server/utils/heapdump.js

```diff
@@ -32,11 +34,12 @@ if (process.execArgv.includes('--report-on-signal')) {
 process.on('SIGUSR2', async () => {
   const targetFilename = `/tmp/heapdump-${process.pid}-${Date.now()}.heapsnapshot`;
   console.warn(`about to write a heapdump to ${targetFilename}`);
+  const heapStream = v8.getHeapSnapshot();
+  const fileStream = fs.createWriteStream(targetFilename);
+
   try {
-    await pipelinePromise(
-      v8.getHeapSnapshot(),
-      fs.createWriteStream(targetFilename)
-    );
+    heapStream.pipe(fileStream);
+    await finishedPromise(fileStream);
```

Comment:
No problem with this solution. I don't see anything to indicate that pipeline changed between 12 and 16, out of curiosity whats the reason for this change ?

Reply:
The promise from `pipeline` wasn't resolving. In node v14 it was updated to only resolve when a `close` event was emitted￼. The option to do that in `createWriteStream` was added in 12 and turned on by default in 14. I did try, as a confidence check, explicitly setting the `emitClose` option to `true`, but it had no effect. I switched to this since we're not doing any transforms so `pipeline` isn't really necessary, and found this implementation didn't have the same issues.